org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -109,9 +109,12 @@ protected void importProjects(Collection<IPath> rootPaths, IProgressMonitor moni
 		SubMonitor subMonitor = SubMonitor.convert(monitor, rootPaths.size() * 100);
 		for (IPath rootPath : rootPaths) {
 			File rootFolder = rootPath.toFile();
-			IProjectImporter importer = getImporter(rootFolder, subMonitor.split(30));
-			if (importer != null) {
+			List<IProjectImporter> importers = new ArrayList<>(importers());
+			IProjectImporter importer = getImporter(rootFolder, importers, subMonitor.split(30));
+			while (importer != null) {
 				importer.importToWorkspace(subMonitor.split(70));
+				importers.remove(importer);
```

Comment:
We actually should remove all importers before and including the current importer. 

Say you have a Maven and Eclipse project in your directory:
1st pass:
- gradle, doesn't apply
- Maven, applies, is removed
- Eclipse, not checked

2nd pass:
- Gradle, retested, still doesn't apply
- Eclipse applies, is removed

3rd pass:
- Gradle retested, still doesn't apply

Reply:
Fixed.